packages/core/src/app/shipping/SingleShippingForm.tsx

```diff
@@ -119,231 +123,190 @@ class SingleShippingForm extends PureComponent<
-                    shippingAddress,
-                ),
-            });
-        }
-    }
-
-    render(): ReactNode {
-        const {
-            cartHasChanged,
-            isInitialValueLoaded,
-            isLoading,
-            onUnhandledError,
-            methodId,
-            shippingAddress,
-            consignments,
-            shouldShowOrderComments,
-            initialize,
-            isValid,
-            deinitialize,
-            values: { shippingAddress: addressForm },
-            isShippingStepPending,
-            shippingFormRenderTimestamp,
-        } = this.props;
-
-        const { isResettingAddress, isUpdatingShippingData, hasRequestedShippingOptions } =
-            this.state;
-
-        const PAYMENT_METHOD_VALID = ['amazonpay'];
-        const shouldShowBillingSameAsShipping = !PAYMENT_METHOD_VALID.some(
-            (method) => method === methodId,
+            shippingAutosaveDelay,
         );
 
-        return (
-            <Form autoComplete="on">
-                <Fieldset>
-                    <ShippingAddress
-                        consignments={consignments}
-                        deinitialize={deinitialize}
-                        formFields={this.getFields(addressForm && addressForm.countryCode)}
-                        hasRequestedShippingOptions={hasRequestedShippingOptions}
-                        initialize={initialize}
-                        isLoading={isResettingAddress}
-                        isShippingStepPending={isShippingStepPending}
-                        methodId={methodId}
-                        onAddressSelect={this.handleAddressSelect}
-                        onFieldChange={this.handleFieldChange}
-                        onUnhandledError={onUnhandledError}
-                        onUseNewAddress={this.onUseNewAddress}
-                        shippingAddress={shippingAddress}
-                    />
-                    {shouldShowBillingSameAsShipping && (
-                        <div className="form-body">
-                            <BillingSameAsShippingField />
-                        </div>
-                    )}
-                </Fieldset>
-
-                <ShippingFormFooter
-                    cartHasChanged={cartHasChanged}
-                    isInitialValueLoaded={isInitialValueLoaded}
-                    isLoading={isLoading || isUpdatingShippingData}
-                    isMultiShippingMode={false}
-                    shippingFormRenderTimestamp={shippingFormRenderTimestamp}
-                    shouldDisableSubmit={this.shouldDisableSubmit()}
-                    shouldShowOrderComments={shouldShowOrderComments}
-                    shouldShowShippingOptions={isValid}
-                />
-            </Form>
-        );
-    }
+        return () => {
+            debouncedUpdateAddressRef.current?.cancel();
+        };
+    }, [updateAddress, shippingAutosaveDelay]);
```

Comment:
```suggestion
    }, []);
```
We only want it to run once, don't we?